src/startup.jl

```diff
@@ -44,8 +44,18 @@ if !symbols_present
     isfile(depfile) || error("PyCall not properly installed. Please run Pkg.build(\"PyCall\")")
     include(depfile) # generated by Pkg.build("PyCall")
     # Only to be used at top-level - pointer will be invalid after reload
-    libpy_handle = Libdl.dlopen(libpython, Libdl.RTLD_LAZY|Libdl.RTLD_DEEPBIND|Libdl.RTLD_GLOBAL)
+    try
+        Libdl.dlopen(libpython, Libdl.RTLD_LAZY|Libdl.RTLD_DEEPBIND|Libdl.RTLD_GLOBAL)
+    catch err
+        if err isa ErrorException
+            error(err.msg, ". Please run `Pkg.build(\"PyCall\")` if your Python build has changed")
+        else
+            rethrow(err)
+        end
+    end
     # need SetPythonHome to avoid warning, #299
+    libpy_handle = Libdl.dlopen(libpython, Libdl.RTLD_LAZY|Libdl.RTLD_DEEPBIND|Libdl.RTLD_GLOBAL)
```

Comment:
Instead of calling `dlopen` twice, you can just do `libpy_handle = try` above, i.e. the library is the return value of the `try` block.

Reply:
Thank you, fixed!